### DISCUSSION TITLE
PostgreSQL Azure EntraID auth

### DISCUSSION THREAD

**Discussion by paulaustin-automutatio** [dbeaver/dbeaver#32347] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I'm using https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-configure-sign-in-azure-ad-authentication instead of password authentication to connect to a postgresql database.

When I started using it I had to every hour request a new password from azure using one of the following commands. Then edit the connection and update the password.

```PowerShell
Get-AzAccessToken -ResourceUrl 'https://ossrdbms-aad.database.windows.net'
```

```bash
az account get-access-token --resource https://ossrdbms-aad.database.windows.net
```

I then simplified it by having a script that updates my pgpass file with the new token and using pgpass authentication in DBeaver.

What would be nice is if Azure Entra ID token auth was implemented in DBeaver

**Describe the solution you'd like**
This simplest solution would be to have an Azure EntraID auth option in the menu. It would then before each connection call the az account... to get the password.

Slightly more advanced would be to cache the token until the expiry of the token.

More advanced would be configuration options and making direct OAuth calls to get the tokens.

**Describe alternatives you've considered**
See first section

**Additional context**
Microsoft does have a java library to do this. However it does the authentication call on every single connection as that is the only way the postgresql JDBC driver allows you to pass in a password provider. In my own code I created a custom data source that uses the cached token as a password, or refeshes when expired.

**Comment by E1izabeth** [comment, marked as answer]
@paulaustin-automutatio this authentication method is available in our PRO product. You can find information about this feature here https://dbeaver.com/docs/dbeaver/Authentication-Azure-AD/

**Reply by djpremier** [reply]
Hi,

Any special reason why this feature cannot be available in the community version?
